extensions/blocks/map/map.php

```diff
@@ -14,6 +14,19 @@
 	)
 );
 
+/**
+ * Return the site's own Mapbox API key if set, or the Automattic's one otherwise.
+ *
+ * @return string
+ */
+function jetpack_get_mapbox_api_key() {
+	if ( ! class_exists( 'WPCOM_REST_API_V2_Endpoint_Service_API_Keys' ) ) {
```

Comment:
I wonder if we should add an additional condition here, for sites that are not connected to WordPress.com? This way we would shortcircuit early when we know a site will not be able to communicate with WordPress.com. What do you think?

Reply:
JP blocks are only available to sites connected to WPCOM, so there should be no need for that condition.

Reply:
One can use some of the Jetpack blocks (like the Map block) when their site is [in development mode](https://jetpack.com/support/development-mode) and has not been connected to WordPress.com, so I think we would need to account for that.